In [14]:
import pandas as pd #dataframe which helps in viewing the data in to tabular format
from sklearn.preprocessing import LabelEncoder #charcyers like Y/N are used to enode inot 0 or 1 to make machine understand
from sklearn.model_selection import train_test_split #to train test data 
from sklearn.linear_model import LinearRegression 
from sklearn.metrics import accuracy_score, confusion_matrix #to calculate accuracy of the model as well as other metrics
from sklearn.metrics import r2_score

In [15]:
df=pd.read_csv("insurance[1].csv")
df

age     sex     bmi  children smoker     region      charges
0      19  female  27.900         0    yes  southwest  16884.92400
1      18    male  33.770         1     no  southeast   1725.55230
2      28    male  33.000         3     no  southeast   4449.46200
3      33    male  22.705         0     no  northwest  21984.47061
4      32    male  28.880         0     no  northwest   3866.85520
...   ...     ...     ...       ...    ...        ...          ...
1333   50    male  30.970         3     no  northwest  10600.54830
1334   18  female  31.920         0     no  northeast   2205.98080
1335   18  female  36.850         0     no  southeast   1629.83350
1336   21  female  25.800         0     no  southwest   2007.94500
1337   61  female  29.070         0    yes  northwest  29141.36030

[1338 rows x 7 columns]

As you can see:
1. dimension of the dataset is 1338 rows and 7 columns
2. the attribute sex and smoker are categorical values either f/m or y/n hence label encoder will be used to convert into binary values 
3. region also has the same concept but it will not convert just into 0 and 1 because it has four regions hence will be turned into 0-3

In [16]:
#preprocessing the data
lb=LabelEncoder()
#for sex
lb.fit(df.sex.drop_duplicates()) #to only pick unique values and convert them
df.sex=lb.transform(df.sex)
#for smoker
lb.fit(df.smoker.drop_duplicates())
df.smoker=lb.transform(df.smoker)
#for region
lb.fit(df.region.drop_duplicates())
df.region=lb.transform(df.region)
df

age  sex     bmi  children  smoker  region      charges
0      19    0  27.900         0       1       3  16884.92400
1      18    1  33.770         1       0       2   1725.55230
2      28    1  33.000         3       0       2   4449.46200
3      33    1  22.705         0       0       1  21984.47061
4      32    1  28.880         0       0       1   3866.85520
...   ...  ...     ...       ...     ...     ...          ...
1333   50    1  30.970         3       0       1  10600.54830
1334   18    0  31.920         0       0       0   2205.98080
1335   18    0  36.850         0       0       2   1629.83350
1336   21    0  25.800         0       0       3   2007.94500
1337   61    0  29.070         0       1       1  29141.36030

[1338 rows x 7 columns]

In [66]:
#splitting into train test
x=df.drop(['charges'],axis=1)
y=df[['charges']]

x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.1,random_state=23) #to split into 70-30% using test size as 0.3
LinearModel=LinearRegression()
LinearModel.fit(x_train,y_train)
pred=LinearModel.predict(x_test)

In [67]:
#to check the coefficient that the model generates and fits to satisfies the eqn
for idx,col_name in enumerate(x_train.columns):
    print("The coeff for {} is {}".format(col_name,LinearModel.coef_[0][idx]))

The coeff for age is 252.3861045864902
The coeff for sex is -124.98954588930854
The coeff for bmi is 343.8766798260682
The coeff for children is 518.5888858269627
The coeff for smoker is 23531.909261549536
The coeff for region is -421.15667230969814


As we can see:
1. The impact of being smoker impacts the model hugely
2. the bmi also effects to an extent

In [71]:
intercept=LinearModel.intercept_[0]
intercept

-11906.468541660739

In [72]:
#predictig the data
#ypred=LinearModel.predict(x_test)
#pred=pd.DataFrame(x_test)
#pred['Actual Cost']=y_test
#pred['Predicted Cost']=ypred
#pred

In [73]:
LinearModel.score(x_test,y_test)

0.8064598346762054